### PR TITLE
[docs] EAS Builds: add 'Android Builds' and 'iOS Builds' pages

### PR DESCRIPTION
**Pull request by dsokal**
# Why

This is the next part of adding docs for EAS Builds.

Previous PRs:

- #9566
- #9613
- #9704

# How

I added new pages for EAS Builds - "Android Builds" and "iOS Builds" which describes both the Android and iOS builds internals.

# Test Plan

I ran the dev server and made sure the new page renders correctly.

### CODE REVIEW

File: docs/pages/build/android-builds.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+## Build Process
+
+Let's take a closer look at the steps for building Android projects with EAS Builds.
+
+### Local Steps
+
+The first phase is happening on your computer, Expo CLI is in charge of completing the following steps:
+
+1. Check if the git index is clean - this means if there aren't any uncommitted changes. If it's not clean an error is thrown.
+2. Prepare the credentials needed for the build unless `builds.android.PROFILE_NAME.withoutCredentials` is set to `true`.
```
Comment (wkozyra95) [inline docs/pages/build/android-builds.md:16]: do we support that? I think project configuration on iOS required credentials

File: docs/pages/build/android-builds.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+2. Download the project tarball from AWS S3 and unpack it.
+3. Run `yarn install` in the project root.
+4. Restore the keystore (if it was included in the build request).
+5. Run `./gradlew COMMAND` in the `android` directory inside your project.
+
+   `COMMAND` is the command defined in your `eas.json` at `builds.android.PROFILE_NAME.gradleCommand`. It defaults to `:app:bundleRelease` which produces the AAB (Android App Bundle).
+
+6. Upload the build artifact to AWS S3.
+
+   The artifact path can be configured in `eas.json` at `builds.android.PROFILE_NAME.artifactPath`. It defaults to `android/app/build/outputs/bundle/release/app-release.aab`.
```
Comment (wkozyra95) [inline docs/pages/build/android-builds.md:45]:
```suggestion
   The artifact path can be configured in `eas.json` at `builds.android.PROFILE_NAME.artifactPath`. It defaults to `android/app/build/outputs/bundle/release/app-release.aab`. For some older projects that default might not be correct and you might need to change that to `android/app/build/outputs/bundle/release/app.aab`.
```

File: docs/pages/build/android-builds.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+## Project Auto-Configuration
+
+Every time you want to build a new Android app binary, we're checking if the project is set up correctly so we can semlessly run the build process on our servers.
+
+### Android Keystore
+
+Android requires you to sign your application with a certificate. That certificate is stored in your keystore. The Google Play store identifies applications based on the certificate. It means that if you lose your keystore you may not be able to update your application in the store. However, with [Play App Signing](https://developer.android.com/studio/publish/app-signing#app-signing-google-play), you can mitigate the risk of losing your keystore.
```
Comment (wkozyra95) [inline docs/pages/build/android-builds.md:53]:
```suggestion
Android requires you to sign your application with a certificate. That certificate is stored in your Keystore. The Google Play store identifies applications based on the certificate. It means that if you lose your Keystore you may not be able to update your application in the store. However, with [app signing by Google Play](https://developer.android.com/studio/publish/app-signing#app-signing-google-play), you can mitigate the risk of losing your keystore.
```

Comment (wkozyra95) [reply docs/pages/build/android-builds.md:53]: `app signing by Google Play` is used in google documentation

File: docs/pages/build/android-builds.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  android.buildTypes.release { config ->
+    config.signingConfig android.signingConfigs.release
+  }
+}
+```
+
+The most important part is the `release` signing config. It's configured to read the keystore and passwords from the `credentials.json` file at the project root. Even though you're not required to create this file on your own, it's created and populated with your credentials by EAS Builds before running the build.
+
```
Comment (wkozyra95) [inline docs/pages/build/android-builds.md:116]:
```suggestion
If you specify `android.signingConfigs.release` in your build.gradle we will not override that configuration. In that case, your local builds will use credentials you configured manually and on the worker instance, we will use either credentials.json or credentials from Expo servers.
```

File: docs/pages/build/android-builds.md (hunk 1 above)
Comment (dsokal) [reply docs/pages/build/android-builds.md:16]: (discussed offline) This page is for Android, not for iOS.

File: docs/pages/build/android-builds.md (hunk 2 above)
Comment (dsokal) [reply docs/pages/build/android-builds.md:45]: I'd prefer to include this information in the `Configuring with eas.json` page.

File: docs/pages/build/android-builds.md (hunk 3 above)
Comment (dsokal) [reply docs/pages/build/android-builds.md:53]: No, it's not. Open the link :)

**Review by wkozyra95 — APPROVED** [review]

File: docs/pages/build/android-builds.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    }
+  }
+
+  android.buildTypes.release { config ->
+    config.signingConfig android.signingConfigs.release
+  }
+}
+```
+
+The most important part is the `release` signing config. It's configured to read the keystore and passwords from the `credentials.json` file at the project root. Even though you're not required to create this file on your own, it's created and populated with your credentials by EAS Builds before running the build. Please note that if `android.signingConfigs.release.storeFile` is already specified in your `build.gradle` we will **not** override your configuration.
```
Comment (brentvatne) [inline docs/pages/build/android-builds.md:115]: we could use the `@info` markup ([like this](https://github.com/expo/expo/blame/master/docs/pages/tutorial/image.md#L24)/[see that live](https://docs.expo.io/tutorial/image/#displaying-the-image-in-the-app)) to add yellow highlights on the parts that are relevant to make it easier to spot

**Review by brentvatne — CHANGES_REQUESTED** [review]

File: docs/pages/build/ios-builds.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
+---
+title: iOS Builds
+---
+
+This page describes the process of building iOS projects with EAS Builds. Dive in if you're interested in the implementation details of our build service.
+
+## Build Process
+
+Let's take a closer look at the steps for building iOS projects with EAS Builds.
```
Comment (brentvatne) [inline docs/pages/build/ios-builds.md:9]:
```suggestion
Let's take a closer look at the steps for building iOS projects with EAS Builds. We first run some steps on your local machine to prepare the project, and then we actually build the project on a remote service.
```

File: docs/pages/build/android-builds.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
+---
+title: Android Builds
+---
+
+This page describes the process of building Android projects with EAS Builds. Dive in if you're interested in the implementation details of our build service.
+
+## Build Process
+
+Let's take a closer look at the steps for building Android projects with EAS Builds.
```
Comment (brentvatne) [inline docs/pages/build/android-builds.md:9]:
```suggestion
Let's take a closer look at the steps for building Android projects with EAS Builds. We first run some steps on your local machine to prepare the project, and then we actually build the project on a remote service.
```

File: docs/pages/build/ios-builds.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+   Depending on the value of `builds.ios.PROFILE_NAME.credentialsSource`, the credentials are obtained from either the local `credentials.json` file or from the Expo servers. If the `auto` or `remote` mode is selected but no credentials exist yet, you're offered to generate them.
+
+4. Configure the Xcode project.
+
+   4.1. Ensure the correct bundle identifier and Apple Team ID are set.
+
+   4.2. Set the id of the provisioning profile resolved in the previous step.
+
+5. Create the tarball containing all your project sources - run `git archive --format=tar.gz --prefix project/ -o project.tar.gz HEAD`.
+6. Upload the project tarball to AWS S3 and send the build request to EAS Builds.
```
Comment (brentvatne) [inline docs/pages/build/ios-builds.md:31]:
```suggestion
6. Upload the project tarball to a private AWS S3 bucket and send the build request to EAS Builds.
```

or whatever we decide from the android section is a better substitute to emphasize secure storage of sources

**Review by brentvatne — CHANGES_REQUESTED** [review]
i love the detail in this, if we keep this up to date it will be a great resource for developers using the service. nice work @dsokal

File: docs/pages/build/android-builds.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+   Depending on the value of `builds.android.PROFILE_NAME.credentialsSource`, the credentials are obtained from either the local `credentials.json` file or from the Expo servers. If the `auto` or `remote` mode is selected but no credentials exist yet, you're offered to generate a new keystore.
+
+3. Check if the Android project is configured so it could be built on the EAS Builds servers.
+
+   Namely, Expo CLI is checking whether `android/app/build.gradle` contains `apply from: "./eas-build.gradle"`.
+   If the project is not configured, run auto-configuration steps ([learn more below](#project-auto-configuration)).
+
+4. Create the tarball containing all your project sources - run `git archive --format=tar.gz --prefix project/ -o project.tar.gz HEAD`.
+5. Upload the project tarball to AWS S3 and send the build request to EAS Builds.
```
Comment (dsokal) [inline docs/pages/build/android-builds.md:26]:
```suggestion
5. Upload the project tarball to a private AWS S3 bucket and send the build request to EAS Builds.
```

File: docs/pages/build/android-builds.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+### Remote Steps
+
+Next, this is what happens when EAS Builds picks up your request:
+
+1. Create a new Docker container for the build.
+
+   Every build gets its own, fresh container with all build tools installed there (Java JDK, Android SDK, NDK, and so on).
+
+2. Download the project tarball from AWS S3 and unpack it.
```
[the remark this answers is not in the document]
Comment (dsokal) [reply docs/pages/build/android-builds.md:36]:
```suggestion
2. Download the project tarball from a private AWS S3 bucket and unpack it.
```

File: docs/pages/build/ios-builds.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+### Remote Steps
+
+Next, this is what happens when EAS Builds picks up your request:
+
+1. Create a new macOS VM for the build.
+
+   Every build gets its own, fresh macOS VM with all build tools installed there (Xcode, fastlane, and so on).
+
+2. Download the project tarball from AWS S3 and unpack it.
```
Comment (dsokal) [inline docs/pages/build/ios-builds.md:41]:
```suggestion
2. Download the project tarball from a private AWS S3 bucket and unpack it.
```

File: docs/pages/build/ios-builds.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+   5.2. Import the Distribution Certificate into the keychain.
+
+   5.3. Write the Provisioning Profile to the `~/Library/MobileDevice/Provisioning Profiles` directory.
+
+   5.4. Verify that the Distribution Certificate and Provisioning Profile match (every Provisioning Profile is assigned to a particular Distribution Certificate and cannot be used for building the iOS with any other certificate).
+
+6. Create `Gymfile` in the `ios` directory if it does **not** exist (check out the [Default Gymfile](#default-gymfile) section).
+7. Run `fastlane gym` in the `ios` directory.
+8. Upload the build artifact to AWS S3.
```
Comment (dsokal) [inline docs/pages/build/ios-builds.md:56]:
```suggestion
8. Upload the build artifact to a private AWS S3 bucket.
```

File: docs/pages/build/android-builds.md (hunk 5 above)
Comment (dsokal) [reply docs/pages/build/android-builds.md:115]: 🤯

Comment (dsokal) [reply docs/pages/build/android-builds.md:115]: I'll try to use it in a separate PR.

File: docs/pages/build/android-builds.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+3. Run `yarn install` in the project root.
+4. Restore the keystore (if it was included in the build request).
```
[the remark this answers is not in the document]
Comment (dsokal) [reply docs/pages/build/android-builds.md:38]: Actually, being managed by Expo is not the only case. A user could have his own `credentials.json` with the keystore defined there. When building a release build - the keystore will be included in the build request but it's not actually managed by Expo.

**Comment by dsokal** [comment]
Like with the previous PR - thanks for the feedback, Brent!

**Pull request merged**